datasets/cmu_mocap/cmu_mocap.py

```diff
@@ -0,0 +1,301 @@
+                        "amc": datasets.Sequence(datasets.Value("string")),
+                        "asf": datasets.Value("string"),
+                    }
+                }
+            )
+        else:
+            features.update(
+                {
+                    "motions": datasets.Sequence(datasets.Value("string")),
+                }
+            )
+
+        return features
+
+    def _get_data_paths(self, data_dir):
+        if self.config.name == "asf-amc":
+            data_paths = [os.path.join(data_dir, "all_asfamc", "subjects")]
+        elif self.config.name == "avi":
+            data_paths = [os.path.join(data_dir, "subjects")]
+        else:
+            data_paths = []
+            for path in data_dir:
+                data_path = os.path.join(path, "subjects")
+                data_paths.append(data_path)
+
+        return data_paths
+
+    def _split_generators(self, dl_manager):
+        urls = _URLS[self.config.name]
+        metadata_path = dl_manager.download(_METADATA_URL)
+        data_dir = dl_manager.download_and_extract(urls)
+        data_paths = self._get_data_paths(data_dir)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "data_paths": data_paths,
+                    "metadata_path": metadata_path,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    @staticmethod
+    def _get_category_subcategory_info(files, subject_id_to_details):
+        categories = []
+        subcategories = []
+        descriptions = []
+
+        for file_ in files:
+            trial_number = os.path.basename(file_).split(".")[0]
+            if trial_number not in subject_id_to_details:
+                category_name = ""
+                sub_category_name = ""
+                description = ""
+            else:
+                category_id = str(subject_id_to_details[trial_number]["main_cat"])
+                category_name = _CATEGORIES[category_id]
+                sub_category_id = str(subject_id_to_details[trial_number]["sub_cat"])
+                sub_category_name = _SUB_CATEGORY_NAMES[(category_id, sub_category_id)]
+                description = subject_id_to_details[trial_number]["description"]
+            categories.append(category_name)
+            subcategories.append(sub_category_name)
+            descriptions.append(description)
+
+        return [categories, subcategories, descriptions]
+
+    @staticmethod
+    def _get_subject_paths(data_paths):
+        def listdir_fullpath(d):
+            return [os.path.join(d, f) for f in os.listdir(d)]
+
+        subject_paths = []
+        for data_path in data_paths:
+            subject_paths.extend(listdir_fullpath(data_path))
+
+        return subject_paths
+
+    def _generate_examples(self, data_paths, metadata_path, split):
```

Comment:
The URLs support streaming from what I see, so it would be cool to make the dataset streamable. Let me know if you need help with that.